Add XML Elements

Lukas595 opened the pull request:
Add XML Elements for
- Add VC to Display Geometry
- Custom Content Space
- Export

Lukas595 committed:
Add XML Elements

Add XML Elements for
- Add VC to Display Geometry
- Custom Content Space
- Export

Ahmed-jerbi reviewed (approved):
Awesome job Lukas! the XML elements are nice and clean.
I noticed you added btnSelectExportDestination_Click to get an export path, 
Another idea is to create one single function that will do that for all path Text boxes. its role will be to simply show the dialogbox and return a string path. Then we attach it as an event to controls whenever we need. This will make our final code shorter and cleaner, do you wanna give it a try?

Lukas595 commented:
Yes, I can do that.

Should I add this to this pull request or open a new branch?

Ahmed-jerbi commented:
> 
> 
> Yes, I can do that.
> 
> Should I add this to this pull request or open a new branch?

You can add it to this pull request, 
Later on we can diverge into different branches when we work simultaneously on more features.

Lukas595 committed:
Update Main.cs

- add `showSelectFolderDialog` function

Lukas595 committed:
Add functions to show save/open file dialog

- starting with input validation
- replace textbox input for 'wait' block with NumericUpDown

Ahmed-jerbi commented on Main.cs:
```diff
@@ -4,13 +4,16 @@
         // Declaring global variables
         public XDocument xdoc = new XDocument(new XDeclaration("1.0", "utf-8",null ),new XElement("VIOSO"));
+
+        private List<String> validationErrorList = new List<string>;
```
This line generates a compilation error

Ahmed-jerbi commented on Main.cs:
```diff
@@ -369,12 +412,57 @@ private void resetButton_Click(object sender, EventArgs e)
+            String destinationFile = String.Empty;
+            if (showSaveFileDialog(ref destinationFile))
+            {
+                srcText.Text = destinationFile;
```
This saves the path in the wrong box

Ahmed-jerbi commented on Main.cs:
```diff
@@ -369,12 +412,57 @@ private void resetButton_Click(object sender, EventArgs e)
+            }
+        }
+
+        private void compoundVCText_Validating(object sender, System.ComponentModel.CancelEventArgs e)
```
Nice little warning trick!

Ahmed-jerbi reviewed (commented):
Hey Lukas! nice job, i see you explored a lot of functions in c# .net 👍 

At this point, I'd like to go back a step behind and focus on the basics of the program: generating correct XML. Later on we can add the browsing functions, input validation, error handling, UI ..etc.

So the essential step is to compare the reference XML to the generated one, I see we have differences in the export for example its not matching correctly. 

A good way to check is also in the VIOSO application, from the menu Calibration > Run script 
With this you can test the generated scripts from our program, and spot any errors 

Let me know how this turns out, and don't hesitate to call me anytime :)

Lukas595 commented:
Thanks for the feedback, i have fixed the things you mentioned. Okay I will roll back the changes for the validation and focus on the XML generation.

Should I then add the functions/XML elements that are missing, such as: PERSPECTIVE CORRECTION FOR STATIC EYE POINT: OBSERVER CORRECTION and CALIBRATION ADJUSTMENTS: MASK AND COLOR etc. ?

Ahmed-jerbi commented:
Let's not add those functions yet, our goal now is to release a first simple Alpha version that has basic few XML elements and is tested to generate proper scripts that run in VIOSO. Once we are there we can continue and improve/add to it.
Did you get a chance to try the VIOSO run script function?

Lukas595 committed:
Update Main.cs

- add XML Comments
- add Docs
- rollback input validation checks

Lukas595 commented:
All right, I understand.
No, sadly I don't have the possibility to run the script from home.

Ahmed-jerbi commented:
You can download the VIOSO VR&SIM software [here](https://vioso.sharepoint.com/:u:/s/intern/EfT5yA02CcxAkptKNGSXLmoBsmySHcH9b6_1r9hoHpBb7A?e=DYoU4b)

This is a trial version, but has full functionalities and just adds a watermark on outputs, so feel free to use it in home office :)

Lukas595 commented:
Good, I will try it out tomorrow

Lukas595 committed:
Dialogue windows now open in the relevant folders

Lukas595 committed:
Update Main.cs

- fix typo
